contracts/interfaces/IAelinPool.sol

```diff
@@ -12,5 +12,17 @@ interface IAelinPool {
         uint256 purchaseDuration;
         address[] allowListAddresses;
         uint256[] allowListAmounts;
+        NftData[] nftData;
+    }
+
+    // collectionAddress should be unique, otherwise will override
+    struct NftData {
+        // if 0, then unlimited purchase
+        uint256 purchaseAmount;
+        address collectionAddress;
+        // if true, then `purchaseAmount` is per token
+        // else `purchaseAmount` is per account regardless of the NFTs held
+        bool purchaseAmountPerToken;
+        // can add tokenId for 1155, but cannot be 0 because there are many 721s with tokenid 0
```

Comment:
the tokenIDs field is a good idea and will only apply to the 1155 collections, not the 721. For 721 we have to assume that all ids are valid

Reply:
only the purchse1155 method will check this tokenIDs field